docs/source/kinds_of_types.rst

```diff
@@ -721,8 +721,8 @@ For more details, see :ref:`type-variable-value-restriction`.
 Generators
 **********
 
-A basic generator that only yields values can be annotated as having a return
-type of either :py:class:`Iterator[YieldType] <typing.Iterator>` or :py:class:`Iterable[YieldType] <typing.Iterable>`. For example:
```

Comment:
I don't agree on this change. Sometimes you can refactor generators into regular function and vice a versa. Imagine this:

```python
def return_values() -> Iterable[int]:
    return [1, 2, 3]
```

Then you refactor it to be a generator instead:

```python
def return_values() -> Iterable[int]:
    yield 1
    yield 2
    yield 3
```

But, the return type can stay the same. 

Or, sometimes generators are not required and measuring performance says that it is faster / cheaper to just use a `list`. So, you do the opposite refactoring.

I often use this as a good practice to annotate functions like this as `Iterable`. I know that they might change. And I don't want to have a really specific return type.

Or, another case: you can design your functions to forbid using `next()` explicitly. It might matter for some generators.

Reply:
Instead I propose to make it more clear: which return type do you need? `Iterable`, `Iterator`, or `Generator`. A table maybe? 🤔

Reply:
I think it's good for mypy to advocate, in general, using as specific a return-type as possible 🙂

If you're writing a function that is to be part of the public API for a library, it's important for other users of your code to know that `next()` can safely be called on the object that's returned from this function. If you refactor `return_values` from a regular function to a generator, I'd argue you *should* change the return-type annotation, as otherwise mypy will raise false-positive errors if a user of your hypothetical library tries to call `next()` on the returned object.

I do see your point, though — I'll have a think about ways I could change this proposed alteration to make it clearer 🙂